src/main/scala/decider/PathConditions.scala

```diff
@@ -142,12 +142,17 @@ private trait LayeredPathConditionStackLike {
   protected def conditionalized(layers: Stack[PathConditionStackLayer]): Seq[Term] = {
     var unconditionalTerms = Vector.empty[Term]
     var conditionalTerms = Vector.empty[Term]
+    var implicationLHS: Term = True()
 
-    for (layer <- layers) {
+    for (layer <- layers.reverseIterator) {
       unconditionalTerms ++= layer.globalAssumptions
 
+      layer.branchCondition foreach { condition =>
+         implicationLHS = And(implicationLHS, condition)
```

Comment:
`And` is n-ary - can't that be exploited to avoid the loop over layer.branchCondition?

Reply:
@mschwerhoff `layer.branchCondition` is an `Option[Term]` so that inner block will execute at most once

Reply:
Yeah that's probably more clear, I've changed it to use a match instead.